test/conftest.py

```diff
@@ -328,37 +328,38 @@ def t_sfdr(classname, devicename, channel, param_set, sfdr_min):
     # Set custom device parameters
     for p in param_set.keys():
         setattr(sdr, p, param_set[p])
-    time.sleep(5)  # Wait for QEC to kick in
+    # time.sleep(5)  # Wait for QEC to kick in
     # Set common buffer settings
-    N = 2 ** 14
     sdr.tx_cyclic_buffer = True
-    sdr.tx_enabled_channels = [channel]
-    sdr.tx_buffer_size = N * 2 * len(sdr.tx_enabled_channels)
-    sdr.rx_enabled_channels = [channel]
-    sdr.rx_buffer_size = N * 2 * len(sdr.rx_enabled_channels)
+    sdr.rx_buffer_size = 2 ** 15
+    if sdr._num_tx_channels > 2:
+        sdr.tx_enabled_channels = [channel]
+    if sdr._num_rx_channels > 2:
+        sdr.rx_enabled_channels = [channel]
```

Comment:
Did the previous line 333-338 break your code? I tried to remove the "if sdr._num_tx_channels > 2:" so that I can enable 1 channel only from a none complex-data device that only have 2 TX channels like DAQ2.

Reply:
So we don't have any SFDR tests for non-complex devices yet, but in that case we might want to implement a single sided SFDR measurement. Then you don't have to use DAC/ADC pairs.